cpp/src/arrow/type.h

```diff
@@ -691,7 +691,7 @@ class ARROW_EXPORT Time64Type : public TimeType {
 
   int bit_width() const override { return static_cast<int>(sizeof(c_type) * CHAR_BIT); }
 
-  explicit Time64Type(TimeUnit::type unit = TimeUnit::MILLI);
+  explicit Time64Type(TimeUnit::type unit = TimeUnit::MICRO);
```

Comment:
Any reasons to change the default here? This is a quite hidden breaking change.

Reply:
I think that `TimeUnit::MILLI` is invalid because https://github.com/apache/arrow/blob/master/cpp/src/arrow/type.h#L112-L114 says:

```cpp
    /// Time as signed 64-bit integer, representing either microseconds or
    /// nanoseconds since midnight
    TIME64,
```

So I think that this is just a bug fix.